### PR TITLE
lab1

### PR DESCRIPTION
**Pull request by VeranikaKemtsava**

### CODE REVIEW

File: StudentPerformance/StudentPerformance/BL/IStudentPerfomanceService.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿using StudentPerformance.BL.Dto;
+using System.Collections.Generic;
+
+namespace StudentPerformance.BL
+{
+    interface IStudentPerfomanceService
+    {
+        public void LoadInfo(string filePath);
```
[the remark this answers is not in the document]
Comment (VeranikaKemtsava) [reply StudentPerformance/StudentPerformance/BL/IStudentPerfomanceService.cs:8]: removed

File: StudentPerformance/StudentPerformance/BL/StudentPerfomanceService.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+using System.Collections.Generic;
+using System.Linq;
+
+namespace StudentPerformance.BL
+{
+    class StudentPerfomanceService : IStudentPerfomanceService
+    {
+        private IPerfomanceDataLoader _loader;
+
+        private IEnumerable<Student> _students;
```
[the remark this answers is not in the document]
Comment (VeranikaKemtsava) [reply StudentPerformance/StudentPerformance/BL/StudentPerfomanceService.cs:13]: fixed

File: StudentPerformance/StudentPerformance/Output/XmlPerfomanceStatisticsSaver.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    class XmlPerfomanceStatisticsSaver : IPerfomanceStatisticsSaver
+    {
+        public void SavePerfomanceStatistics(string filePath,
+            IEnumerable<StudentAverageMark> studentAverageMarks, IEnumerable<SubjectAverageMark> subjectAverageMarks, double averageMark)
+        {
+            using var p = new ExcelPackage();
+
+            var ws = p.Workbook.Worksheets.Add("Average marks");
+
+            ws.Cells[1, 1].Value = "Фамилия";
```
[the remark this answers is not in the document]
Comment (VeranikaKemtsava) [reply StudentPerformance/StudentPerformance/Output/XmlPerfomanceStatisticsSaver.cs:17]: done

File: StudentPerformance/StudentPerformance/Output/XmlPerfomanceStatisticsSaver.cs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+            var area = ws.Cells[2, 1].LoadFromCollection(studentAverageMarks);
+
+            ws.Cells[area.End.Row + 1, area.Start.Column].Value = "Cредняя оценка по группе";
+            ws.Cells[area.End.Row + 1, area.End.Column].Value = averageMark;
+
+            ws.Cells[1, 6].Value = "Предмет";
+            ws.Cells[1, 7].Value = "Средняя оценка";
+            ws.Cells[2, 6].LoadFromCollection(subjectAverageMarks);
+            
+            p.SaveAs(new FileInfo(filePath + ".xlsx"));
```
[the remark this answers is not in the document]
Comment (VeranikaKemtsava) [reply StudentPerformance/StudentPerformance/Output/XmlPerfomanceStatisticsSaver.cs:30]: fixed

File: StudentPerformance/StudentPerformance/Output/IPerfomanceStatisticsSaver.cs [hunk 5]
```diff
@@ -0,0 +1,13 @@
+﻿using StudentPerformance.BL.Dto;
+using System.Collections.Generic;
+
+namespace StudentPerformance.Output
+{
+    interface IPerfomanceStatisticsSaver
+    {
+        public void SavePerfomanceStatistics(string filePath,
```
[the remark this answers is not in the document]
Comment (VeranikaKemtsava) [reply StudentPerformance/StudentPerformance/Output/IPerfomanceStatisticsSaver.cs:8]: done

File: StudentPerformance/StudentPerformance/DAL/IPerfomanceDataLoader.cs [hunk 6]
```diff
@@ -0,0 +1,13 @@
+﻿using StudentPerformance.DAL.Models;
+using System.Collections.Generic;
+
+namespace StudentPerformance.DAL
+{
+    interface IPerfomanceDataLoader
+    {
+        public void LoadPerfomanceData(string filePath,
```
[the remark this answers is not in the document]
Comment (VeranikaKemtsava) [reply StudentPerformance/StudentPerformance/DAL/IPerfomanceDataLoader.cs:8]: done

File: StudentPerformance/StudentPerformance/Output/JsonPerfomanceStatisticsSaver.cs [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+using System.Text;
+
+namespace StudentPerformance.Output
+{
+    class JsonPerfomanceStatisticsSaver : IPerfomanceStatisticsSaver
+    {
+        public void SavePerfomanceStatistics(string filePath,
+            IEnumerable<StudentAverageMark> studentAverageMarks, IEnumerable<SubjectAverageMark> subjectAverageMarks, double averageMark)
+        {
+            using var writer = new StreamWriter(filePath + ".json", false, Encoding.UTF8);
```
[the remark this answers is not in the document]
Comment (VeranikaKemtsava) [reply StudentPerformance/StudentPerformance/Output/JsonPerfomanceStatisticsSaver.cs:14]: fixed

File: StudentPerformance/StudentPerformance/Program.cs [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+
+        [Required]
+        [Option("-f", "Format of output file (json, excel)", CommandOptionType.SingleValue)]
+        public string Format { get; }
+
+        private void OnExecute()
+        {
+            Log.Logger = new LoggerConfiguration()
+                .WriteTo.Console()
+                .WriteTo.File("log.txt")
```
[the remark this answers is not in the document]
Comment (VeranikaKemtsava) [reply StudentPerformance/StudentPerformance/Program.cs:33]: fixed

File: StudentPerformance/StudentPerformance/Program.cs [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+            Log.Information("App started.");
+
+            var services = new ServiceCollection()
+                .AddTransient<App>()
+                .AddTransient<IStudentPerfomanceService, StudentPerfomanceService>()
+                .AddTransient<IPerfomanceDataLoader, CsvPerfomanceDataLoader>()
+                .AddLogging(c => c.AddSerilog());
+            switch (Format.ToLower())
+            {
+                case "excel":
```
[the remark this answers is not in the document]
Comment (VeranikaKemtsava) [reply StudentPerformance/StudentPerformance/Program.cs:44]: replaced with enum

File: StudentPerformance/StudentPerformance/Program.cs [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+
+        private void OnExecute()
+        {
+            Log.Logger = new LoggerConfiguration()
+                .WriteTo.Console()
+                .WriteTo.File("log.txt")
+                .CreateLogger();
+            Log.Information("App started.");
+
+            var services = new ServiceCollection()
```
[the remark this answers is not in the document]
Comment (VeranikaKemtsava) [reply StudentPerformance/StudentPerformance/Program.cs:37]: moved

File: StudentPerformance/StudentPerformance/DAL/Models/Mark.cs [hunk 11]
```diff
@@ -0,0 +1,24 @@
+﻿using System;
+
+namespace StudentPerformance.DAL.Models
+{
+    class Mark
```
[the remark this answers is not in the document]
Comment (VeranikaKemtsava) [reply StudentPerformance/StudentPerformance/DAL/Models/Mark.cs:5]: can i leave it as it is?

[the remark this answers is not in the document]
Comment (alexey62soft) [reply StudentPerformance/StudentPerformance/DAL/Models/Mark.cs]: Okay, let's leave as is)

**Pull request merged**